### PR TITLE
Add task of AliAnalysisTaskEmcalSubjet

### PR DESCRIPTION
**Pull request by xcheung**
- Add task of AliAnalysisTaskEmcalSubjet
  - Location: PWGJE/EMCALJetTasks/UserTasks
  - Usage: sub-jet analysis with ALICE data
- Modify property of fFillGhost in AliEmcalJetTask
  - Allowing to save its prompt value in root file
- Changes to be committed:
  - modified: PWGJE/EMCALJetTasks/AliEmcalJetTask.h
  - modified: PWGJE/EMCALJetTasks/CMakeLists.txt
  - modified: PWGJE/EMCALJetTasks/PWGJEEMCALJetTasksLinkDef.h
  - new file: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalSubjet.cxx
  - new file: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalSubjet.h

### CODE REVIEW

**Comment by raymondEhlers** [comment]
+test

File: PWGJE/EMCALJetTasks/AliEmcalJetTask.h [hunk 1]
```diff
@@ -189,7 +189,7 @@ class AliEmcalJetTask : public AliAnalysisTaskEmcal {
   Bool_t                 fIsPSelSet;              //!<!=true if physics selection was set
   Bool_t                 fIsEmcPart;              //!<!=true if emcal particles are given as input (for clusters)
   Bool_t                 fLegacyMode;             //!<!=true to enable FJ 2.x behavior
-  Bool_t                 fFillGhost;              //!<!=true ghost particles will be filled in AliEmcalJet obj
+  Bool_t                 fFillGhost;              ///< =true ghost particles will be filled in AliEmcalJet obj
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/AliEmcalJetTask.h]: This change needs to be accompanied by an increase in the ClassDef

Comment (xcheung) [reply PWGJE/EMCALJetTasks/AliEmcalJetTask.h]: @raymondEhlers Dear Raymond: thanks for point it out. The version number was changed. See my new commit

**Comment by aiola** [comment]
+1

**Pull request merged**